### ISSUE TITLE
Add RPM packaging

### ISSUE THREAD

**Issue by Dahu88** [crmarsh/discord-linux-bugs#24]
It may be early in the beta phase but I have friends who'd like to try Discord on a Fedora and don't want to install it by hand.
So the RPM packaging should be helpful.

**Comment by TheDoctorsLife** [comment]
It should be as easy as extracting the tar.gz

**Comment by crmarsh** [comment]
Yeah, I'll take a look at more packaging/distribution stuff when this hits stable.

**Comment by ahjolinna** [comment]
it would also help other distros like arch/gentoo (+their spinoffs), if there was a list of all the needed dependencies

**Comment by crmarsh** [comment]
For the deb control file I have
```
Depends: libc6, libasound2, libgconf-2-4, libnotify4, libnspr4, libnss3, libstdc++6, libxss1, libxtst6, libappindicator1
```
I dunno how those translate into arch/gentoo/whatever package names.

**Comment by ahjolinna** [comment]
this is my dependencies list for [chakraOS](https://chakraos.org) : `alsa-lib atk bzip2 cairo dbus-glib e2fsprogs expat fontconfig freetype2 gconf gdk-pixbuf2 gmp gnutls graphite harfbuzz keyutils libcap libcups libdatrie libdbus libdrm libffi libgcrypt libgpg-error libnotify libpng libtasn1 libthai libxau libxcomposite libxcursor libxdamage libxdmcp libxext libxfixes libxi libxinerama libxrandr libxrender libxshmfence libxtst libxxf86vm lz4 p11-kit pcre pixman systemd xz zlib`

**Comment by ahjolinna** [comment]
one of my scipts says that I have been using says that only these deps should be enough:  `alsa-lib dbus-glib gconf gtk2 libnotify libxtst nss` but for some reason my friend (who had fresh install of chakraos) couldn't get his mic to work, he did work for him with that other longer deps list

**Comment by TheDoctorsLife** [comment]
This is what my script says in needed for runtime. Where is says discord-canary is there becuase I would ship those (becuase I'm to lazy to make packages for them).
```bash
   SONAME: libstdc++.so.6 <-> libstdc++>=4.4.0_1
   SONAME: libpthread.so.0 <-> glibc>=2.8_1
   SONAME: libc.so.6 <-> glibc>=2.8_1
   SONAME: libnode.so <-> discord-canary (ignored)
   SONAME: libdiscord.so <-> discord-canary (ignored)
   SONAME: libXss.so.1 <-> libXScrnSaver>=1.1.3_1
   SONAME: libgtk-x11-2.0.so.0 <-> gtk+>=2.16.0_1
   SONAME: libgdk-x11-2.0.so.0 <-> gtk+>=2.16.0_1
   SONAME: libatk-1.0.so.0 <-> atk>=1.26.0_1
   SONAME: libpangocairo-1.0.so.0 <-> pango>=1.24.0_1
   SONAME: libcairo.so.2 <-> cairo>=1.8.6_1
   SONAME: libpango-1.0.so.0 <-> pango>=1.24.0_1
   SONAME: libfreetype.so.6 <-> freetype>=2.3.9_1
   SONAME: libfontconfig.so.1 <-> fontconfig>=2.6.0_1
   SONAME: libnotify.so.4 <-> libnotify>=0.7_1
   SONAME: libgdk_pixbuf-2.0.so.0 <-> gdk-pixbuf>=2.22.0_1
   SONAME: libgio-2.0.so.0 <-> glib>=2.18.0_1
   SONAME: libgobject-2.0.so.0 <-> glib>=2.18.0_1
   SONAME: libdbus-1.so.3 <-> dbus-libs>=1.2.10_1
   SONAME: libXi.so.6 <-> libXi>=1.2.1_1
   SONAME: libXcursor.so.1 <-> libXcursor>=1.1.9_1
   SONAME: libXdamage.so.1 <-> libXdamage>=1.1.1_1
   SONAME: libXrandr.so.2 <-> libXrandr>=1.3.0_1
   SONAME: libXcomposite.so.1 <-> libXcomposite>=0.4.0_1
   SONAME: libXext.so.6 <-> libXext>=1.0.5_1
   SONAME: libXfixes.so.3 <-> libXfixes>=4.0.3_1
   SONAME: libXrender.so.1 <-> libXrender>=0.9.4_1
   SONAME: libX11.so.6 <-> libX11>=1.2_1
   SONAME: libXtst.so.6 <-> libXtst>=1.0.3_1
   SONAME: libgconf-2.so.4 <-> GConf>=2.32.0_1
   SONAME: libgmodule-2.0.so.0 <-> glib>=2.18.0_1
   SONAME: librt.so.1 <-> glibc>=2.8_1
   SONAME: libglib-2.0.so.0 <-> glib>=2.18.0_1
   SONAME: libnss3.so <-> nss>=3.12.4_1
   SONAME: libnssutil3.so <-> nss>=3.12.4_1
   SONAME: libsmime3.so <-> nss>=3.12.4_1
   SONAME: libnspr4.so <-> nspr>=4.8_1
   SONAME: libffmpeg.so <-> discord-canary (ignored)
   SONAME: libasound.so.2 <-> alsa-lib>=1.0.20_1
   SONAME: libcups.so.2 <-> libcups>=1.5.3_1
   SONAME: libdl.so.2 <-> glibc>=2.8_1
   SONAME: libexpat.so.1 <-> expat>=2.0.0_1
   SONAME: libm.so.6 <-> glibc>=2.8_1
   SONAME: libgcc_s.so.1 <-> libgcc>=4.4.0_1
   SONAME: ld-linux-x86-64.so.2 <-> glibc>=2.8_1
```

**Comment by fishxz** [comment]
why not just http://flatpak.org/? all problems solved...

**Comment by hickeroar** [comment]
The solution to this issue seems implicit. Naturally when the app goes gold it's going to be distributed via apt/dpkg/yum/rpm. In a Beta or Alpha, no one should really expect extra convenience stuff to be added to the distribution. My $0.02 :)

**Comment by crmarsh** [comment]
Flatpak does not appear to solve the problem of targeting multiple distros, it looks like it is basically another package manager to target.

**Comment by fishxz** [comment]
i just gonna quote the webiste:  

**Flatpak is the new framework for desktop applications on Linux**  
Distributing applications on Linux is a pain: different distributions in multiple versions, each with their own versions of libraries and packaging formats. Flatpak is here to change all that. It allows the same app to be installed on different Linux distributions, including different versions. And it has been designed from the ground up with security in mind, so that apps are isolated from each other and from the host system.

**Comment by crmarsh** [comment]
[Right](https://xkcd.com/927/)

**Comment by fishxz** [comment]
then stop complaining about different packagemanger and how to target them.  
im out.

**Comment by TheDoctorsLife** [comment]
[opensuse's build system](https://build.opensuse.org/) has a lot of the packaging for multiple distros automated, should be a easy solution to take care of multiple distrobutions.

**Comment by TheElectronWill** [comment]
I agree that a proper RPM package would be helpful... so I've just made one!
-> [https://github.com/RPM-Outpost/discord](https://github.com/RPM-Outpost/discord)
Tested on Fedora 24.

**Comment by ahjolinna** [comment]
there is something new called : [PackPack](https://github.com/packpack/packpack): "[Simple Building Of RPMs & Debian Packages From Git Repos](http://phoronix.com/scan.php?page=news_item&px=PackPack-Deb-RPM-Building)"

thought I would like to see Flatpak/snap version also, I personally prefer just a Flatpak as its XDG standard and not some canonical uncertain standard (like upstart and MIR -_-)

**Comment by CorruptComputer** [comment]
@crmarsh Now that Discord for Linux is officially released, could we get this? I regularly use Fedora/openSUSE and it is a pain to use Discord in its current state. Also nothing against @TheElectronWill but I don't like using 3rd party apps, as I have had my passwords stolen in the past.

**Comment by TheElectronWill** [comment]
@CorruptComputer I understand your concerns about security, but I'd like to point out that what I've created are just two bash scripts and some other text files (.spec and .desktop). They're not very long so you can read them and check easily that they're not evil.

**Comment by CorruptComputer** [comment]
@TheElectronWill I can see that's all you have and that they are not dangerous, however I would still feel better with official support.

**Comment by TheElectronWill** [comment]
@CorruptComputer I agree that an official RPM would be better.
@crmarsh Any plan to do it?

**Comment by crmarsh** [comment]
I've got it on my todo list (alongside making a snap package), but they have both been there for a while.

**Comment by hevanaa** [comment]
FYI there is an unofficial Fedora repo for Discord Canary that many seem to be using (including me) https://copr.fedorainfracloud.org/coprs/vishalv/discord-canary/

**Comment by CorruptComputer** [comment]
@crmarsh I don't know if this helps any, but Snap is a universal Linux package. So you wouldn't have to worry about distro support, it'll handle all that for you. https://snapcraft.io

**Comment by hevanaa** [comment]
@CorruptComputer If you must advertise Snap, there is also a flatpak available at https://flathub.org/apps.html. Not for Canary, though.
